pkg/controller/remoteingress/remoteingress_controller.go

```diff
@@ -169,62 +273,79 @@ func newSyncSetSpec(cd *hivev1.ClusterDeployment, rawExtensions []runtime.RawExt
 	return ssSpec
 }
 
-func (r *ReconcileRemoteClusterIngress) syncSyncSet(cd *hivev1.ClusterDeployment, rawExtensions []runtime.RawExtension) error {
+// syncSyncSet builds up a syncSet object with the passed-in rawExtensions as the spec.Resources
+func (r *ReconcileRemoteClusterIngress) syncSyncSet(rContext *reconcileContext, rawExtensions []runtime.RawExtension) error {
+	ssName := fmt.Sprintf("%v-clusteringress", rContext.clusterDeployment.Name)
 
-	ssName := fmt.Sprintf("%v-clusteringress", cd.Name)
-	newSyncSetSpec := newSyncSetSpec(cd, rawExtensions)
-
-	ss := &hivev1.SyncSet{}
-	err := r.Get(context.TODO(), types.NamespacedName{Name: ssName, Namespace: cd.Namespace}, ss)
-	if errors.IsNotFound(err) {
-		// create the syncset
-		ss = &hivev1.SyncSet{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      ssName,
-				Namespace: cd.Namespace,
-			},
-			Spec: *newSyncSetSpec,
-		}
+	newSyncSetSpec := newSyncSetSpec(rContext.clusterDeployment, rawExtensions)
+	syncSet := &hivev1.SyncSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ssName,
+			Namespace: rContext.clusterDeployment.Namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "SyncSet",
+			APIVersion: hivev1.SchemeGroupVersion.String(),
+		},
+		Spec: *newSyncSetSpec,
+	}
 
-		// ensure the syncset gets cleaned up when the clusterdeployment is deleted
-		if err := controllerutil.SetControllerReference(cd, ss, r.scheme); err != nil {
-			r.logger.WithError(err).Error("error setting owner reference")
-			return err
-		}
+	// ensure the syncset gets cleaned up when the clusterdeployment is deleted
+	if err := controllerutil.SetControllerReference(rContext.clusterDeployment, syncSet, r.scheme); err != nil {
+		r.logger.WithError(err).Error("error setting owner reference")
+		return err
+	}
 
-		if err := r.Create(context.TODO(), ss); err != nil {
-			r.logger.Errorf("error creating sync set: %v", err)
-			return err
-		}
+	kubeCLI, err := r.kubeCLIBuilder(r.restConfig, rContext.logger)
+	if err != nil {
+		rContext.logger.WithError(err).Error("error building kubeCLI client")
+	}
 
-		return nil
-	} else if err != nil {
-		r.logger.Errorf("error checking for existing syncset: %v", err)
+	serializer := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
```

Comment:
fyi ... in my branch I'm changing the signature of the function to take a scheme and then just build the serializer in the implementation. But hopefully yours merges first and then I'll just update it here as well.